src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -397,6 +397,18 @@ export class OnlineGameWrapperComponent extends GameWrapper<MinimalUser> impleme
         return this.requestManager.canMakeRequest('Draw');
     }
 
+    public override async canUserPlay(_clickedElementName: string): Promise<MGPValidation> {
+        const result: MGPValidation = await super.canUserPlay(_clickedElementName);
+        if (result.isFailure()) {
+            return result;
+        }
+        if (this.mustReply()) {
```

Comment:
hm en terme d'ordre c'est pas mieux l'inverse ? Si tu dois répondre, peu importe où tu clique, ça dit « tu dois répondre », sinon ça fait le super

Reply:
le où tu clique n'est pas encore une question, l'ordre de prio (de facto)
CANNOT_PLAY_AS_OBSERVER
NOT_YOUR_TURN
GAME_HAS_ENDED
Et on ajoute, en ligne,
MUST_ANSWER_REQUEST

Fun que "CANNOT_PLAY_AS_OBSERVER" soit pas strictement réservé à OGWC, je le bougerais pas ?

là où on utilise canUserPlay il est utilisé au début et si il a déjà qqch à dire, on le dit direct, là j'ai suivi le pattern.
Et il y aura plus de clics lors ce que c'est pas ton tour ou en tant qu'observateur, le code derrière must-reply est prévu dans un contexte ou t'as été prouvé comme non-observateur, ton tour, jeu en cours, let us keep it that ouais, innit mate ?

Reply:
Ah oui selon cet ordre là c'est bien à cet endroit alors, ça le ferait pas qu'en tant qu'observateur tu aies un "non non tu DOIS répondre"
Et oui, je suis pour que CANNOT_PLAY_AS_OBSERVER soit seulement dans OGWC

Reply:
Donc OGWC a
1. CANNOT_PLAY_AS_OBSERVER
2. NOT_YOUR_TURN
3. GAME_HAS_ENDED
4. MUST_ANSWER_REQUEST

C'correc lo ?

Reply:
ouaip ! Peut-être juste mettre un petit commentaire pour pas oublier (surtout que le GW va checker not your turn & game has ended seulement)